From bf03deb7cab468c68799de338ba2819aef9bdd83 Mon Sep 17 00:00:00 2001
From: YuZhoujian <windyu@tencent.com>
Date: Tue, 24 Mar 2020 20:25:17 +0800
Subject: [PATCH] bdev/nvme: Reduce the long time for apply_nvme_firmware RPC

nvme_adminq_poll_period_us is set to 1000000 us by default, which
is much too infrequent for cases like, downloading firmware. So
reduce nvme_adminq_poll_period_us to 100000 us.

$ time ./scripts/rpc.py apply_firmware ./xxx_signed.bin
632f5018-0c5c-49b7-9aa9-aba41291c3d3-nvmen1
"firmware commit succeeded. Controller reset in progress."

real    0m36.209s
user    0m0.056s
sys     0m0.008s

Signed-off-by: YuZhoujian <windyu@tencent.com>
---
 module/bdev/nvme/bdev_nvme.c | 2 +-
 scripts/config_converter.py  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/module/bdev/nvme/bdev_nvme.c b/module/bdev/nvme/bdev_nvme.c
index 8590b58d3..c446152d3 100644
--- a/module/bdev/nvme/bdev_nvme.c
+++ b/module/bdev/nvme/bdev_nvme.c
@@ -116,7 +116,7 @@ static struct spdk_bdev_nvme_opts g_opts = {
 	.low_priority_weight = 0,
 	.medium_priority_weight = 0,
 	.high_priority_weight = 0,
-	.nvme_adminq_poll_period_us = 1000000ULL,
+	.nvme_adminq_poll_period_us = 100000ULL,
 	.nvme_ioq_poll_period_us = 0,
 	.io_queue_requests = 0,
 	.delay_cmd_submit = SPDK_BDEV_NVME_DEFAULT_DELAY_CMD_SUBMIT,
diff --git a/scripts/config_converter.py b/scripts/config_converter.py
index caaed5e66..5d933ab75 100755
--- a/scripts/config_converter.py
+++ b/scripts/config_converter.py
@@ -206,7 +206,7 @@ def get_nvme_bdev_json(config, section):
     params = [
         ["RetryCount", "retry_count", int, 4],
         ["TimeoutuSec", "timeout_us", int, 0],
-        ["AdminPollRate", "nvme_adminq_poll_period_us", int, 1000000],
+        ["AdminPollRate", "nvme_adminq_poll_period_us", int, 100000],
         ["ActionOnTimeout", "action_on_timeout", str, "none"],
         ["IOPollRate", "nvme_ioq_poll_period_us", int, 0],
         ["HotplugEnable", "enable", bool, False],
-- 
2.23.0.37.g745f681

